source/solvers/vof_algebraic_interface_reinitialization.cc

```diff
@@ -0,0 +1,835 @@
+// SPDX-FileCopyrightText: Copyright (c) 2020-2024 The Lethe Authors
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
+
+#include <core/solutions_output.h>
+
+#include <solvers/vof_algebraic_interface_reinitialization.h>
+
+#include <deal.II/lac/solver_control.h>
+#include <deal.II/lac/sparsity_tools.h>
+#include <deal.II/lac/trilinos_solver.h>
+
+#include <sys/stat.h>
+
+template <int dim>
+void
+VOFAlgebraicInterfaceReinitialization<dim>::setup_dofs()
+{
+  // Get MPI communicator
+  auto mpi_communicator = this->triangulation->get_communicator();
+
+  // Distribute and renumber DoFs
+  this->dof_handler.distribute_dofs(*this->fe);
+  DoFRenumbering::Cuthill_McKee(this->dof_handler);
+
+  // Get locally owned and relevant DoFs
+  this->locally_owned_dofs = this->dof_handler.locally_owned_dofs();
+  this->locally_relevant_dofs =
+    DoFTools::extract_locally_relevant_dofs(this->dof_handler);
+
+  // Constraints
+  define_non_zero_constraints();
+  define_zero_constraints();
+
+  // Sparsity pattern
+  DynamicSparsityPattern dsp(this->locally_relevant_dofs);
+  DoFTools::make_sparsity_pattern(this->dof_handler,
+                                  dsp,
+                                  this->nonzero_constraints,
+                                  true);
```

Comment:
```suggestion
                                  false);
```
You don't need to keep the constrainted DOFs I believe. The defauilt behavior is false